hedera-node/src/main/java/com/hedera/services/store/models/Topic.java

```diff
@@ -0,0 +1,166 @@
+package com.hedera.services.store.models;
+
+/*
+ * -
+ * ‌
+ * Hedera Services Node
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.hedera.services.exceptions.InvalidTransactionException;
+import com.hedera.services.legacy.core.jproto.JKey;
+import com.hedera.services.state.submerkle.RichInstant;
+import com.hederahashgraph.api.proto.java.ConsensusCreateTopicTransactionBody;
+import com.hederahashgraph.api.proto.java.Key;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import org.apache.commons.codec.DecoderException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.time.Instant;
+
+/**
+ * Represents the model of a {@link com.hedera.services.state.merkle.MerkleTopic}.
+ * 
+ * @author Yoan Sredkov
+ */
+public class Topic {
+	private final static Logger log = LogManager.getLogger(Topic.class);
+	
+	private final Id id;
+	private String memo;
+	private JKey adminKey;
+	private JKey submitKey;
+	private Id autoRenewAccountId;
+	private long autoRenewDurationSeconds;
+	private boolean deleted;
+	private boolean isNew;
+	private RichInstant expirationTimestamp;
+	
+	private long sequenceNumber;
+	private byte[] runningHash;
+	
+ 	public Topic(final Id id) {
+		this.id = id;
+	}
+
+	/**
+	 * Creates a new {@link Topic} from the given body.
+	 * Note: The created model is not added to state, and must be explicitly persisted via {@link com.hedera.services.store.TopicStore#persistNew(Topic)}
+	 * 
+	 * @param body - the gRPC transaction body
+	 * @param id - the id generated in the transition logic
+	 * @param expirationTime - expiration time of the topic,
+	 *               or when {@link com.hedera.services.txns.consensus.SubmitMessageTransitionLogic} will start failing.
+	 * @return - the new topic
+	 */
+	public static Topic fromGrpcTopicCreate(ConsensusCreateTopicTransactionBody body, Id id, Instant expirationTime) {
```

Comment:
I think that it would be better to pass the variables in the method instead of passing the whole GRPC body. Here are the reasons why:
- We will cut the usage of GRPC at the TransitionLogic and not propage it in the model
- The properties are 4-5 so they will not be that many
- We can move `attemptDecodeOrThrow` to the options validator class (where I think it should be) and we can parse the keys in the transition logic from the GRPC Key.

`autoRenewAccount`, `submitKey` and `adminKey` can be `@Nullable`